### PR TITLE
Make it easier to register SwiftMailer plugins

### PR DESCRIPTION
**Pull request by c960657**
This patch makes it easier to register SwiftMailer plugins.

### CODE REVIEW

File: tests/Silex/Tests/Provider/SpoolStub.php [hunk 1]
```diff
@@ -36,7 +36,7 @@ public function isStarted()
 
     public function queueMessage(\Swift_Mime_Message $message)
     {
-        $this->messages[] = $message;
+        $this->messages[] = clone $message;
```
Comment (fabpot) [inline tests/Silex/Tests/Provider/SpoolStub.php]: Should probably be part of another PR as it does not seems related and not tested.

File: src/Silex/Provider/SwiftmailerServiceProvider.php [hunk 2]
```diff
@@ -97,13 +86,34 @@ public function register(Container $app)
...
+            }
+
+            foreach ($plugins as $plugin) {
+                $dispatcher->bindEventListener($plugin);
+            }
+
+            return $dispatcher;
         };
 
+        $app['swiftmailer.plugins'] = array();
```
Comment (fabpot) [inline src/Silex/Provider/SwiftmailerServiceProvider.php]: This should not be an array as per your example in the docs, the plugins would not be lazy-loaded. Instead, you should do the same as for `validator.object_initializers` for instance:

```php
$app['swiftmailer.plugins'] = function ($app) {
    return array();
}
```

Comment (c960657) [inline src/Silex/Provider/SwiftmailerServiceProvider.php]: Fixed.

File: tests/Silex/Tests/Provider/SpoolStub.php (hunk 1 above)
Comment (c960657) [inline tests/Silex/Tests/Provider/SpoolStub.php]: Without this, the new `testSwiftMailerSenderAddress` fails.

[Swift_Plugins_ImpersonatePlugin](https://github.com/swiftmailer/swiftmailer/blob/master/lib/classes/Swift/Plugins/ImpersonatePlugin.php) reverts the headers after the message has been sent (in the `sendPerformed` event), so without cloning, `$message->getReturnPath()` would not return the value used when the message is sent.

File: src/Silex/Provider/SwiftmailerServiceProvider.php (hunk 2 above)
Comment (fabpot) [inline src/Silex/Provider/SwiftmailerServiceProvider.php]: Does not look like it is fixed :)

File: tests/Silex/Tests/Provider/SpoolStub.php (hunk 1 above)
Comment (fabpot) [inline tests/Silex/Tests/Provider/SpoolStub.php]: ok

File: src/Silex/Provider/SwiftmailerServiceProvider.php (hunk 2 above)
Comment (c960657) [inline src/Silex/Provider/SwiftmailerServiceProvider.php]: Arg! Fixed "again" :-)

**Comment by fabpot** [comment]
Thank you @c960657.
